alacritty_terminal/src/term/mod.rs

```diff
@@ -1838,45 +1857,86 @@ impl<T: EventListener> Handler for Term<T> {
     #[inline]
     fn clear_screen(&mut self, mode: ansi::ClearMode) {
         trace!("Clearing screen: {:?}", mode);
-        let template = self.grid.cursor.template;
 
-        // Remove active selections.
-        self.selection = None;
+        let template = self.grid.cursor.template;
+        let num_cols = self.grid.num_cols();
+        let num_lines = self.grid.num_lines();
+        let cursor = self.grid.cursor.point;
+        let buffer_cursor = self.grid.buffer_cursor();
 
         match mode {
             ansi::ClearMode::Above => {
-                let cursor = self.grid.cursor.point;
+                let selection_range = self.selection.as_ref().and_then(|s| s.to_range(&self));
```

Comment:
If we accept to only do line-based granularity instead of cell-based, I think we should be fine with just using the selection without expansion to range?

Since the expansion is done on rendering anyways, it will just expand to the newly written/cleared content which doesn't seem that incorrect to me compared to what we currently do.